src/uct/base/uct_component.c

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2019.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "uct_component.h"
+
+#include <ucs/debug/assert.h>
+#include <ucs/debug/memtrack.h>
+#include <ucs/sys/string.h>
+#include <limits.h>
+#include <string.h>
+
+
+ucs_status_t uct_query_components(uct_component_h **components_p,
+                                  unsigned *num_components_p)
+{
+    uct_component_h *components;
+    uct_md_component_t *mdc;
+    size_t num_components;
+
+    num_components = ucs_list_length(&uct_md_components_list);
+    components = ucs_malloc(num_components * sizeof(*components),
+                            "uct_components");
+    if (components == NULL) {
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    ucs_assert(num_components < UINT_MAX);
```

Comment:
it's done here because the line after that we are assigning a size_t variable to unsigned l-value, since this assertion will blow everything up i don't think it really matters we allocate something beforehand